packages/workspace/src/browser/workspace-uri-contribution.ts

```diff
@@ -57,7 +62,7 @@ export class WorkspaceUriLabelProviderContribution extends DefaultUriLabelProvid
         if (!icon) {
             try {
                 const stat = await this.getStat(element);
-                if (stat.isDirectory) {
+                if (stat && stat.isDirectory) {
```

Comment:
I think this is not correct. In general, and it applies to the PR:
 - With the previous code; if the resource did not exist, at least it was rejected, and we had the chance to see the error. That was not so good either.
 - With the current changes; we swallow the rejections and the errors, and we silently fall back to the `else` branch. Here we would show a file icon for missing resources.

Did I get this correctly?

Reply:
You're right, this changes the behavior and we get no error... I can keep the same behavior as before, by manually throwing if the stat is undefined, although like you said, it might not be the best option... Any ideas for this?

Reply:
Maybe we should throw and make sure the error is caught above? Just logging it would infer that I have to change the return type to allow undefined for this method I think.

Reply:
I missed the fallback `return` statement from the `!stat` branch.